core/storage/exploration/gae_models_test.py

```diff
@@ -316,3 +332,90 @@ def test_get_at_least_editable(self):
             exploration_models.ExpSummaryModel
             .get_at_least_editable('nonexistent_id'))
         self.assertEqual(0, len(exploration_summary_models))
+
+
+class StateIdMappingModelUnitTest(test_utils.GenericTestBase):
+    """Tests for the StateIdMappingModel."""
+
+    def test_create_successfully(self):
+        exploration_models.StateIdMappingModel.create(
+            exp_id='id_0',
+            exp_version=0,
+            state_names_to_ids={},
+            largest_state_id_used=0)
+
+        observed_model = (
+            exploration_models.StateIdMappingModel.
+            get_state_id_mapping_model('id_0', 0))
+
+        self.assertEqual(observed_model.state_names_to_ids, {})
+        self.assertEqual(observed_model.largest_state_id_used, 0)
+
+    def test_create_failed_if_id_exists_and_no_overwrite(self):
+        exploration_models.StateIdMappingModel.create(
+            exp_id='id_1',
+            exp_version=0,
+            state_names_to_ids={},
+            largest_state_id_used=1,
+            overwrite=False)
```

Comment:
Interesting, I didn't realize there was an overwrite parameter. Shouldn't you have separate tests for the overwrite=True case and the overwrite=False case?

Reply:
It throws an exception only when overwrite is False.
The code in gae_models: 
`if not overwrite and cls.get_by_id(instance_id):`
            `raise Exception(`
                `'State id mapping model already exists for exploration %s,'`
               `' version %d' % (exp_id, exp_version))`

Reply:
Yep, so, you could have this test for the exception, but you could additionally have a separate test with similar setup and doing a call with overwrite=True. You can then check that the contents of the model instance have changed accordingly.